pennylane/ops/op_math/controlled_decompositions.py

```diff
@@ -127,47 +134,21 @@ def _decompose_recursive(op, power, control_wires, target_wire, work_wires):
     return decomposition
 
 
-def _decompose_mcx_with_many_workers(control_wires, target_wire, work_wires):
+def _decompose_mcx_with_many_workers_old(control_wires, target_wire, work_wires):
     """Decomposes the multi-controlled PauliX gate using the approach in Lemma 7.2 of
     https://arxiv.org/abs/quant-ph/9503016, which requires a suitably large register of
     work wires"""
-    num_work_wires_needed = len(control_wires) - 2
-    work_wires = work_wires[:num_work_wires_needed]
 
-    work_wires_reversed = list(reversed(work_wires))
-    control_wires_reversed = list(reversed(control_wires))
+    with qml.queuing.AnnotatedQueue() as q:
+        wires = list(control_wires) + [target_wire]
+        _decompose_mcx_with_many_workers(wires=wires, work_wires=work_wires)
 
-    gates = []
+    current_queue = qml.queuing.QueuingManager.active_context()
+    if current_queue is not None:
+        for op in q.queue:  # pragma: no cover
+            qml.queuing.apply(op, context=current_queue)
 
-    for i in range(len(work_wires)):
-        ctrl1 = control_wires_reversed[i]
-        ctrl2 = work_wires_reversed[i]
-        t = target_wire if i == 0 else work_wires_reversed[i - 1]
-        gates.append(qml.Toffoli(wires=[ctrl1, ctrl2, t]))
-
-    gates.append(qml.Toffoli(wires=[*control_wires[:2], work_wires[0]]))
-
-    for i in reversed(range(len(work_wires))):
-        ctrl1 = control_wires_reversed[i]
-        ctrl2 = work_wires_reversed[i]
-        t = target_wire if i == 0 else work_wires_reversed[i - 1]
-        gates.append(qml.Toffoli(wires=[ctrl1, ctrl2, t]))
-
-    for i in range(len(work_wires) - 1):
-        ctrl1 = control_wires_reversed[i + 1]
-        ctrl2 = work_wires_reversed[i + 1]
-        t = work_wires_reversed[i]
-        gates.append(qml.Toffoli(wires=[ctrl1, ctrl2, t]))
-
-    gates.append(qml.Toffoli(wires=[*control_wires[:2], work_wires[0]]))
-
-    for i in reversed(range(len(work_wires) - 1)):
-        ctrl1 = control_wires_reversed[i + 1]
-        ctrl2 = work_wires_reversed[i + 1]
-        t = work_wires_reversed[i]
-        gates.append(qml.Toffoli(wires=[ctrl1, ctrl2, t]))
-
-    return gates
+    return q.queue
```

Comment:
What happened to this method? It's named `_old` but using the new one actually? lol